### DISCUSSION TITLE
Play-session playlists

### DISCUSSION THREAD

**Discussion by smileyhead** [ppy/osu#13049] [Ideas]
**Describe the new feature:** You know how a lot of people like to play osu! by pressing the Random button a lot? This would be like that, but a little smarter.
The idea is to have a new menu where you can configure your session. You could tell the game how long you'd like to play; if you want to improve your scores, play unplayed maps, or just whatever; and specify the difficulty spectrum (maybe with an option to arrange the maps from easiest to hardest so the maps get harder as you play).
After inputting your parameters, the game will arrange you a playlist of maps and you can start playing. On the results screen after each map, you would be able to see the entire playlist and after the results screen, the game would start playing the next song immediately instead of exiting to the song carousel. Also, have the option in the pause menu to skip the current song.

**EDIT:**
Additional thoughts I've had since creating this suggestion:
- It could be called Marathon Mode.
- Ability to sort based on any piece of metadata (star rating, BPM, length, etc.), with ascending/descending toggle.
- Log combo chains across multiple maps (a.k.a. don't reset to 0 on map change).
- Implementing [igontusuwu's suggestions](https://github.com/ppy/osu/issues/5757#issuecomment-723223363) as more options:
  - Lives system.
  - Endless mode.
  - Harder HP drain for each map.
  - Disable skipping.
  - Skip the results screen between songs and only show an overlay for a few seconds with your current progress (to help keep you in the flow). Show full results screen at the very end.
  - Retry from the very beginning.
- All of the above should be just options you can turn on or off.

**Proposal designs of the feature:** 
![image](https://user-images.githubusercontent.com/15932372/63226915-a7967000-c1e0-11e9-8587-6da9c6e90356.png)
![image](https://user-images.githubusercontent.com/15932372/63226918-b11fd800-c1e0-11e9-9a76-ed95bf3f2000.png)

**Comment by iSimpFor** [comment]
since the other proposal got closed but i still like the idea of a "how long can you last until you die" mode, i'll just move my idea here as a mode

infinite/endless challenge
-length setting is ignored or locked to infinity (the game will choose maps(that slowly go up in difficulty) for you as you play)
-hp drain becomes harsher for every map passed
-the skipping option is locked
-game skips results screen for all maps until you fail, then the above applies but with additional stats such as survival  
-retry option resets the whole challenge, making you start from the beginning

**Comment by smileyhead** [comment]
@igontusuwu Your proposal might stray just a bit too far from mine in my opinion, but I can definitely see the appeal of a lives system (set to one in your case), setting the length to endless, and an option to disable skipping.
**EDIT:** Also, forcing a harder HP drain might make some (especially harder) maps impossible to complete.

**Comment by iSimpFor** [comment]
> 
> 
> @igontusuwu Your proposal might stray just a bit too far from mine in my opinion, but I can definitely see the appeal of a lives system (set to one in your case), setting the length to endless, and an option to disable skipping.
> **EDIT:** Also, forcing a harder HP drain might make some (especially harder) maps impossible to complete.

it's not going to increase that quickly, the hp drain should be beatmaps passed*0.025 added on to the original hp which will take a while to notice
additionally, the hp drain alone can't kill you since you need a poor hit to fail
